### ISSUE TITLE
Redundant transactions in blocks in the same layer

### ISSUE THREAD

**Issue by lrettig** [spacemeshos/protocol#23]
Following on our conversation in [this thread](https://github.com/spacemeshos/protocol/pull/20#discussion_r373875277):

> barakshani
> we get redundancy, but how much would depend on the number of transactions waiting in the mempool. Given the number of (average) blocks in a layer and the limit of transaction per block (so basically given the limit of transactions in a layer), there is some number of transactions in the mempool for which we don't expect to get duplications at all (if the limit of transactions in a layer is n, then once there are roughly n^2 transactions in the mempool, we expect no duplications)

@barakshani what happens when there is < _n_ transactions in the mempool? Don't all of the blocks in a layer basically contain the same set of transactions then? Nothing would break, but it just seems like a lot of redundant data storage doesn't it?

**Comment by barakshani** [comment]
yes. correct. we may have dynamic tx selection, so miners won't feel there entire block with all the txs (just because of this redundancy)
